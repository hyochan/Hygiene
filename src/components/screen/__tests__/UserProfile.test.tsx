import 'react-native';

import React, { ReactElement } from 'react';
import { RenderAPI, cleanup, render } from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Screen from '../UserProfile';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: ReactElement;
let testingLib: RenderAPI;

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps({
      route: {
        params: {
          updateUser: jest.fn(),
        },
      },
    });

    component = createTestElement(<Screen {...props} />);
    testingLib = render(component);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('Interaction', () => {
  beforeEach(() => {
    props = createTestProps({
      route: {
        params: {
          updateUser: jest.fn(),
        },
      },
    });

    component = createTestElement(<Screen {...props} />);
    testingLib = render(component);
  });

  afterEach(cleanup);

  it('should simulate onClick', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    // const btn = testingLib.queryByTestId('btn');
    // act(() => {
    //   fireEvent.press(btn);
    //   fireEvent.press(btn);
    // });
    // expect(cnt).toBe(3);
  });
});
