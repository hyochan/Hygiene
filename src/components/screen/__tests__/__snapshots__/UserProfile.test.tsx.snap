// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction should simulate onClick 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "flexDirection": "column",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginTop": 40,
                "opacity": 1,
              }
            }
          >
            <View>
              <Image
                source={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "borderRadius": 35,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
              <Image
                source={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 26,
                    "position": "absolute",
                    "right": 0,
                    "width": 26,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginHorizontal": 22,
                  "marginTop": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  },
                  null,
                ]
              }
            >
              Display name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Please write your display name"
              placeholderTextColor="#808080"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginBottom": 40,
                  "marginHorizontal": 22,
                  "marginTop": 28,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  },
                  null,
                ]
              }
            >
              Self introduction
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={true}
              multiline={true}
              numberOfLines={5}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Please introduce yourself"
              placeholderTextColor="#808080"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 16,
                    "height": 200,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;

exports[`Rendering renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "flexDirection": "column",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginTop": 40,
                "opacity": 1,
              }
            }
          >
            <View>
              <Image
                source={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "borderRadius": 35,
                    "height": 70,
                    "width": 70,
                  }
                }
              />
              <Image
                source={
                  Object {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "height": 26,
                    "position": "absolute",
                    "right": 0,
                    "width": 26,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginHorizontal": 22,
                  "marginTop": 40,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  },
                  null,
                ]
              }
            >
              Display name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={true}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Please write your display name"
              placeholderTextColor="#808080"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 16,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                Object {
                  "marginBottom": 40,
                  "marginHorizontal": 22,
                  "marginTop": 28,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#b9b9c4",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "marginBottom": 5,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 14,
                    "fontWeight": "bold",
                  },
                  null,
                ]
              }
            >
              Self introduction
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={true}
              multiline={true}
              numberOfLines={5}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Please introduce yourself"
              placeholderTextColor="#808080"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "fontSize": 15,
                    "fontWeight": "500",
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                  Object {
                    "color": "black",
                    "fontSize": 16,
                    "height": 200,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": "100%",
                  },
                  undefined,
                  Object {
                    "borderBottomWidth": 0.6,
                    "borderColor": "#eaeaf9",
                  },
                  null,
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
